src/client/mod.rs

```diff
@@ -25,39 +24,58 @@ pub mod builder {
 }
 
 
-pub struct Client {
-    transport: ImapTransport,
-    state: ClientState,
+pub fn connect(server: &str) -> io::Result<ImapConnectFuture> {
+    let addr = (server, 993).to_socket_addrs()?.next().ok_or_else(|| {
+        io::Error::new(io::ErrorKind::Other, format!("no IP addresses found for {}", server))
+    })?;
+    Ok(ImapConnectFuture::TcpConnecting(TcpStream::connect(&addr), server.to_string()))
 }
 
-impl Client {
-    pub fn connect(server: &str, handle: &Handle) -> ConnectFuture {
-        let addr = format!("{}:993", server);
-        let addr = addr.to_socket_addrs().unwrap().next().unwrap();
-        let stream = TcpStream::connect(&addr, handle);
-        ConnectFuture::TcpConnecting(stream, server.to_string())
-    }
+pub trait ImapClient {
+    type Original;
```

Comment:
What is the purpose of `Original` here? Why can't you just use `Self`?

Reply:
Ah, that shows I still don't fully grok the trait system. Nice catch, I'll fix it up.